ext/pdo/pdo_dbh.c

```diff
@@ -221,10 +221,52 @@ static char *dsn_from_uri(char *uri, char *buf, size_t buflen) /* {{{ */
 }
 /* }}} */
 
-/* {{{ */
-PHP_METHOD(PDO, __construct)
+
+#define MAX_PDO_SUB_CLASSES 64
+static unsigned int number_of_pdo_driver_class_entries = 0;
+static pdo_driver_class_entry *pdo_driver_class_entries[MAX_PDO_SUB_CLASSES];
+
+// It would be possible remove this and roll it into the standard driver class entries
+// I chose not to do it at this time, as that would break existing PDO extensions
+zend_result pdo_register_driver_specific_class(pdo_driver_class_entry *driver_class_entry)
+{
+	if (number_of_pdo_driver_class_entries >= MAX_PDO_SUB_CLASSES) {
+		php_error_docref(NULL, E_ERROR, "Too many PDO driver subclasses.");
+		return FAILURE;
+	}
+
+	for (int i=0; i<number_of_pdo_driver_class_entries; i+=1) {
```

Comment:
As ``number_of_pdo_driver_class_entries`` is now unsigned.
```suggestion
	for (unsigned int i = 0; i < number_of_pdo_driver_class_entries; i+=1) {
```